### PR TITLE
rm: if vm in pod is still present, wait it stop before remove resource

### PR DESCRIPTION
**Pull request by gnawux**
- use lock protect unset pod.vm
- when delete, if vm is present, set pod status to be None
- when vm stopped, if status is None, reclaim the resource
- when list, do not display the removed pods

Signed-off-by: Xu Wang <gnawux@gmail.com>

### CODE REVIEW

File: daemon/rm.go [hunk 1]
```diff
@@ -36,19 +36,30 @@ func (daemon *Daemon) CleanPod(podId string) (int, string, error) {
...
-	daemon.DeleteVolumeId(podId)
-	code = types.E_OK
 
-	return code, cause, nil
+	return daemon.RemovePodResource(pod)
+}
+
+func (daemon *Daemon) RemovePodResource(p *Pod) (int, string, error) {
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "services", p.id))
+	os.RemoveAll(path.Join(utils.HYPER_ROOT, "hosts", p.id))
```
Comment (feiskyer) [inline daemon/rm.go]: Is pod.cleanupEtcHosts() needed here?

Comment (gnawux) [inline daemon/rm.go]: it is called when pod stopped.

Comment (gnawux) [inline daemon/rm.go]: https://github.com/hyperhq/hyperd/pull/301/files#diff-cfb48529ee9117af9bd2e69c80e66291R25

every time pod stopped, this method will be called. No matter whether remove the pod.

Comment (feiskyer) [inline daemon/rm.go]: Y, thanks.

File: daemon/rm.go [hunk 2]
```diff
@@ -36,19 +36,30 @@ func (daemon *Daemon) CleanPod(podId string) (int, string, error) {
...
+
+	daemon.db.DeletePod(p.id)
+	daemon.RemovePod(p.id)
+	if p.status.Type != "kubernetes" {
+		daemon.RemovePodContainer(p)
+	}
+	daemon.DeleteVolumeId(p.id)
+	code := types.E_OK
+
+	return code, "", nil
```
Comment (gao-feng) [inline daemon/rm.go]: return E_OK here

Comment (gnawux) [inline daemon/rm.go]: ok, this is just copied from the original code.  

will update this, and are there any other comments.

**Comment by gao-feng** [comment]
seems like StopPod(WithinLock) holds the lock of pod and then set the p.vm to nil first, then the PodStopped will not cleanup resource, should we stop setting p.vm to nil in StopPod, leave this job to PodStopped.

**Comment by gnawux** [comment]
@gao-feng, I agree with you, set `p.vm` to `nil` only in `podStopped`. will update this pr in couple of minutes.

**Comment by gao-feng** [comment]
LGTM

**Pull request merged**